pkg/plugins/golang/v2/scaffolds/api.go

```diff
@@ -92,8 +79,15 @@ func (s *apiScaffolder) newUniverse() *model.Universe {
 }
 
 func (s *apiScaffolder) scaffold() error {
-	if s.doResource {
-		s.config.UpdateResources(s.resource.GVK())
+	// Check if we need to do the API and controller before updating with previously existing info
+	doAPI := s.resource.API != nil && s.resource.API.Version != ""
+	doController := s.resource.Controller
```

Comment:
Why create a local env that is == s.resource.Controller? Has no reason for that. Could you please revert the changes and keep it as before?